Jenkinsfile

```diff
@@ -1,6 +1,77 @@
+/**
+ * Build a docker image given the name
+ *
+ * @param image_name the name of the image
+ * @return image the built docker image
+ */
+def build_image(image_name) {
+    def image
+    withDockerServer([uri: PCIC_DOCKER]) {
+        image = docker.build(image_name)
+    }
+
+    return image
+}
+
+
+/**
+ * If the master branch has been tagged we also add the `latest` tag.  Otherwise
+ * we just use the branch name as the tag.
+ *
+ * @return tags a list of the tags for the image
+ */
+def get_tags() {
+    String tag = sh (script: 'git tag --contains', returnStdout: true).trim()
```

Comment:
While, *in practice* we usually only tag a commit once with a single version, git technically allows an arbitrary number of tags.

```
james@basalt:/tmp/foo$ git log
commit afdc173e87cdfc8cbf4d4c184f2d4da5ac6bcd33 (HEAD -> master, tag: some-other-name, tag: 1.0.0-dev, tag: 1.0.0)
Author: James Hiebert <james@hiebert.name>
Date:   Thu Dec 5 09:14:15 2019 -0800

    First commit
james@basalt:/tmp/foo$ git tag --contains
1.0.0
1.0.0-dev
some-other-name
```

Since you're returning a list already, I think you just need to process the output of `git tag` properly to allow for multiple tags. Then you should be good to go.

Reply:
That isn't something I considered, good catch!